WatchGuard/watchguard-epdr/ingest/parser.yml

```diff
@@ -0,0 +1,118 @@
+        value_sep: ":"
+        item_sep: ","
+
+  - name: parsed_date
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.date}}"
+        output_field: datetime
+    filter: "{{parsed_event.message.date != None}}"
+
+  - name: set_ecs_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          event.kind: "event"
+          event.dataset: "Security logs"
+          observer.vendor: "WatchGuard"
+          observer.product: "WatchGuard EPDR"
+          event.module: "watchguard.epdr"
+
+          event.reason: "{{parsed_event.message.network_attack}}"
+
+          "@timestamp": "{{parsed_date.datetime}}"
+
+          file.hash.md5: "{{parsed_event.message.hash}}"
+
+          host.name: "{{parsed_event.message.host_name}}"
+          host.domain: "{{parsed_event.message.domain}}"
+          host.ip: "{{parsed_event.message.local_ip_address}}"
+          client.ip: "{{parsed_event.message.remote_ip_address}}"
+
+          cloud.account.id: "{{parsed_event.message.account_id}}"
+
+          user.name: "{{parsed_event.message.user_name}}"
+
+          source.ip: "{{parsed_event.message.source_ip}}"
+          source.user.name: "{{parsed_event.message.source_user}}"
+
+          threat.technique.name: "{{parsed_event.message.exploit_technique}}"
+          threat.indicator.confidence: "{{parsed_event.message.rule_risk}}"
+          threat.software.alias: "{{parsed_event.message.alias}}"
+
+          #          threat.tactic.id: "{{parsed_event.message.rule_mitre.tactic}}"
+          #          threat.technique.id: "{{parsed_event.message.rule_mitre.technique}}"
+
+          rule.id: "{{parsed_event.message.rule_id}}"
+          rule.name: "{{parsed_event.message.rule_name}}"
+
+          watchguard.epdr.malware.category: "{{parsed_event.message.malware_category}}"
+          watchguard.epdr.malware.type: "{{parsed_event.message.malware_type}}"
+          watchguard.epdr.site.id: "{{parsed_event.message.site_id}}"
+          watchguard.epdr.site.name: "{{parsed_event.message.site_name}}"
+
+      - set:
+          threat.tactic.id: "{{parsed_event.message.rule_mitre[0].tactic or parsed_rule_mitre.message.get('tactic')}}"
+          threat.technique.id: "{{parsed_event.message.rule_mitre[0].technique or parsed_rule_mitre.message.get('tactic')}}"
+        filter: "{{parsed_event.message.get('rule_mitre') != None}}"
+
+      # Device types based on
+      # https://www.watchguard.com/help/docs/API/Content/en-US/endpoint_security/WES_endpoint_security/v1/WES_endpoint_security.html#Retrieve_Security_Events_for_Devices_..184
+      - translate:
+        dictionary:
+          1: "Workstation"
+          2: "Laptop"
+          3: "Server"
+          4: "Mobile device"
+        mapping:
+          parsed_event.message.device_type: watchguard.epdr.device.type
+
+      - set:
+          file.path: "{{parsed_event.message.path.split('|')[-1]}}"
```

Comment:
In addition, I suggest extracting the `[-2]` part of the path in a custom field (`SYSTEMDRIVE`, `DESKTOPDIRECTORY`, ...)